File: javascript/example-bot.js
/**
 * CorvyBot Example - Echo Bot
 * This bot echoes back any message that includes the command prefix.
 */

const CorvyBot = require('./corvy-sdk');

// Replace these values with your actual bot token and API URL
const BOT_TOKEN = 'your_bot_token_here';
const API_URL = 'https://corvy.chat/api/v1';

// Define commands that the bot will respond to
const commands = [
  {
    prefix: '!echo',
    handler: (message) => {
      // Extract the content after the command
      const content = message.content.substring(message.content.indexOf('!echo') + 5).trim();
      return content ? `Echo: ${content}` : 'You said nothing!';
    }
  },
  {
    prefix: '!hello',
    handler: (message) => {
      return `Hello, ${message.user.username}! How are you today?`;
    }
  },
  {
    prefix: '!help',
    handler: () => {
      return 'Available commands: !echo [text], !hello, !help';
    }
  }
];

// Create and start the bot
const bot = new CorvyBot({
  apiToken: BOT_TOKEN,
  apiBaseUrl: API_URL,
  commands: commands
});

console.log('Starting echo bot...');
bot.start(); 

File: elixir/example_bot.ex
defmodule EchoBot do
  @moduledoc """
  CorvyBot Example - Echo Bot
  This bot echoes back any message that includes the command prefix.
  """

  require Logger

  def start do
    # Replace these values with your actual bot token and API URL
    config = %{
      api_token: "your_api_token_here",
      api_base_url: "https://corvy.chat/api/v1",
      commands: [
        %{
          prefix: "!echo",
          handler: fn message ->
            content = message["content"]
            |> String.downcase()
            |> String.split("!echo")
            |> List.last()
            |> String.trim()

            if content == "", do: "You said nothing!", else: "Echo: #{content}"
          end
        },
        %{
          prefix: "!hello",
          handler: fn message ->
            "Hello, #{message["user"]["username"]}! How are you today?"
          end
        },
        %{
          prefix: "!help",
          handler: fn _message ->
            "Available commands: !echo [text], !hello, !help"
          end
        }
      ]
    }

    # Start the SDK directly
    case CorvyBot.SDK.start_link(config) do
      {:ok, pid} ->
        Logger.info("Bot started successfully")
        # Keep the process running
        Process.link(pid)
        :ok
      {:error, reason} ->
        Logger.error("Failed to start bot: #{inspect(reason)}")
        {:error, reason}
    end
  end
end

# Start the bot and keep it running
EchoBot.start()
# Keep the main process alive
Process.sleep(:infinity) 